test/engine/TextIndexScanForWordTest.cpp

```diff
@@ -21,49 +22,142 @@ namespace {
 std::string kg =
     "<a> <p> \"he failed the test\" . <a> <p> \"testing can help\" . <a> <p> "
     "\"some other sentence\" . <b> <p> \"the test on friday was really hard\" "
-    ". <b> <x2> <x> . <b> <x2> <xb2> .";
+    ". <b> <x2> <x> . <b> <x2> <xb2> . <Astronomer> <is-a> <job> .";
+
+std::string wordsFileContent =
+    "astronomer\t0\t1\t1\n"
+    "<Astronomer>\t1\t1\t0\n"
+    "scientist\t0\t1\t1\n"
+    "field\t0\t1\t1\n"
+    "astronomy\t0\t1\t1\n"
+    "astronomer\t0\t2\t0\n"
+    "<Astronomer>\t1\t2\t0\n"
+    ":s:firstsentence\t0\t2\t0\n"
+    "scientist\t0\t2\t0\n"
+    "field\t0\t2\t0\n"
+    "astronomy\t0\t2\t0\n"
+    "astronomy\t0\t3\t1\n"
+    "concentrates\t0\t3\t1\n"
+    "studies\t0\t3\t1\n"
+    "specific\t0\t3\t1\n"
```

Comment:
Had problems implementing it as a lambda function, since if the lambda function gets defined in the anonymous namespace it can't be used directly after. Therefore I defined it in the respective helper file and then used it as requested.